pkg/cmd/cli/describe/projectstatus.go

```diff
@@ -581,3 +569,94 @@ func describeServicePorts(spec kapi.ServiceSpec) string {
 		return " ports " + strings.Join(pairs, ", ")
 	}
 }
+
+func loadServices(g osgraph.Graph, graphLock sync.Mutex, namespace string, kclient kclient.Interface, client client.Interface) error {
```

Comment:
I figured out why I don't like this structure - loading a list of services into the graph in parallel is not the thing we should be optimizing for.  The extra complexity of these methods (passing the locks in, the wait groups) is a direct result of that.  Don't try to be clever with single threaded execution - use parallel.Run to get the graph data in parallel, then add each in turn.  This structure is just too complex to justify.